components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/WorkflowsApiServiceImpl.java

```diff
@@ -0,0 +1,232 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ * /
+ */
+
+package org.wso2.carbon.apimgt.rest.api.publisher.v1.impl;
+
+import com.google.gson.Gson;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.api.APIAdmin;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.api.APIProvider;
+import org.wso2.carbon.apimgt.api.model.API;
+import org.wso2.carbon.apimgt.api.model.Workflow;
+import org.wso2.carbon.apimgt.impl.APIAdminImpl;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.dao.ApiMgtDAO;
+import org.wso2.carbon.apimgt.impl.utils.APIUtil;
+import org.wso2.carbon.apimgt.impl.workflow.*;
+import org.wso2.carbon.apimgt.rest.api.common.RestApiCommonUtil;
+import org.wso2.carbon.apimgt.rest.api.common.RestApiConstants;
+import org.apache.cxf.jaxrs.ext.MessageContext;
+import org.wso2.carbon.apimgt.rest.api.publisher.v1.WorkflowsApiService;
+import org.wso2.carbon.apimgt.rest.api.publisher.v1.dto.WorkflowDTO;
+import org.wso2.carbon.apimgt.rest.api.publisher.v1.dto.WorkflowListDTO;
+import org.wso2.carbon.apimgt.rest.api.publisher.v1.utils.WorkflowMappingUtil;
+import org.wso2.carbon.apimgt.rest.api.util.utils.RestApiUtil;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Set;
+
+import static org.wso2.carbon.utils.multitenancy.MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+
+public class WorkflowsApiServiceImpl implements WorkflowsApiService {
+
+    private static final Log log = LogFactory.getLog(WorkflowsApiService.class);
+    private static final String createdStatus = "CREATED";
+
+    /**
+     * This is used to get the workflow pending request according to ExternalWorkflowReference
+     *
+     * @param externalWorkflowRef is the unique identifier for workflow request
+     * @param messageContext      Message context of the request
+     * @return
+     */
+    @Override
+    public Response workflowsExternalWorkflowRefGet(String externalWorkflowRef, MessageContext messageContext) {
+        try {
+            String tenantDomain = RestApiCommonUtil.getLoggedInUserTenantDomain();
+            APIAdmin apiAdmin = new APIAdminImpl();
+            Workflow workflow = apiAdmin.getworkflowReferenceByExternalWorkflowReferenceID(externalWorkflowRef,
+                    createdStatus, tenantDomain);
+            return Response.ok().entity(WorkflowMappingUtil.fromWorkflowsToInfoDTO(workflow)).build();
+        } catch (APIManagementException e) {
+            RestApiUtil.handleInternalServerError("Error while retrieving workflow request by the " +
```

Comment:
avoid having catch and throw up with error code.

Reply:
Refactored